src/NuGet.Services.Validation.Orchestrator/BaseValidationMessageHandler.cs

```diff
@@ -149,7 +159,28 @@ private async Task<bool> CheckValidatorAsync(CheckValidatorData message)
                 validationSet.PackageKey,
                 validationSet.ValidationTrackingId))
             {
-                await ProcessValidationSetAsync(entity, validationSet, scheduleNextCheck: false);
+                var lease = await TryAcquireLeaseAsync(validationSet.PackageId, validationSet.PackageNormalizedVersion);
+                if (lease.State == LeaseContextState.Unavailable)
+                {
+                    // If the lease is unavailable, we can drop this message. This part of the handler is processing a
+                    // queue-back message. We don't need to retry these since there is the main retry loop also occurring in
+                    // parallel.
+                    _logger.LogInformation(
+                        "The lease {ResourceName} is unavailable. Dropping message for validation set " +
```

Comment:
This log message may be confusing without additional context. Prefer being a little more verbose:

```suggestion
                        "The lease {ResourceName} is unavailable. Dropping check validator message for validation set " +
```